src/stash/src/postgres.rs

```diff
@@ -628,12 +643,13 @@ impl Stash {
                 // can't accidentally have the same epoch, nonce pair (especially risky if the
                 // current epoch has been bumped exactly once, then gets recreated by another
                 // connection that also bumps it once).
-                let row = tx
-                    .query_one(
-                        "UPDATE fence SET epoch=epoch+1, nonce=$1 RETURNING epoch",
-                        &[&self.nonce.to_vec()],
-                    )
-                    .await?;
+                let update = match epoch_lower_bound {
+                    Some(epoch_lower_bound) => {
+                        format!("UPDATE fence SET epoch=GREATEST(epoch+1, {}), nonce=$1 RETURNING epoch", epoch_lower_bound.get())
```

Comment:
(I have no opinion on the single query idea.)

But I do think this should not use string formatting and instead change params to a vec that can have varying sizes if needed.

Reply:
> change params to a vec that can have varying sizes if needed.

I don't think I fully understand what this means. Does https://github.com/MaterializeInc/materialize/pull/24290/commits/bd133024b18f55fe59dcf5642dc206f548406113 accomplish this?

Reply:
Yes.